core/modules/catalog/helpers/index.ts

```diff
@@ -105,13 +105,20 @@ export function attributeImages (product) {
     for (let attribute of config.products.gallery.imageAttributes) {
       if (product[attribute]) {
         attributeImages.push({
-          'src': getThumbnailPath(product[attribute], config.products.gallery.width, config.products.gallery.height),
-          'loading': getThumbnailPath(product[attribute], 310, 300),
-          'error': getThumbnailPath(product[attribute], 310, 300)
+          src: getThumbnailPath(product[attribute], config.products.gallery.width, config.products.gallery.height),
+          loading: getThumbnailPath(product[attribute], config.products.thumbnails.width, config.products.thumbnails.height),
+          error: getThumbnailPath(product[attribute], config.products.thumbnails.width, config.products.thumbnails.height)
         })
       }
     }
   }
+  if (!attributeImages.length) {
+    attributeImages.push({
+      src: getThumbnailPath(product.image, config.products.gallery.width, config.products.gallery.height),
+      loading: getThumbnailPath(product.image, config.products.thumbnails.width, config.products.thumbnails.height),
+      error: getThumbnailPath(product.image, config.products.thumbnails.width, config.products.thumbnails.height)
```

Comment:
If we don't have `product.image` then maybe we should use `config.images.productPlaceholder`?
```
      src: getThumbnailPath(config.images.productPlaceholder, config.products.gallery.width, config.products.gallery.height),
      loading: getThumbnailPath(config.images.productPlaceholder, config.products.thumbnails.width, config.products.thumbnails.height),
      error: getThumbnailPath(config.images.productPlaceholder, config.products.thumbnails.width, config.products.thumbnails.height)
```

Reply:
Also lets add code to `getProductGallery` because there we build gallery and adding placeholder is last step (we are adding it only if there are no images)

Reply:
I have updated the code by adding the default placeholder condition to the `getProductGallery` and removing the condition to add the default placeholder in the attributeImages().

@gibkigonzo and @Fifciu Please review.

Reply:
looks good for me :+1: